src/main/java/uk/gov/pay/directdebit/payers/resources/PayerResource.java

```diff
@@ -22,27 +25,30 @@
 
 @Path("/")
 public class PayerResource {
-    private static final String PAYERS_API_PATH = "/v1/api/accounts/{accountId}/payment-requests/{paymentRequestExternalId}/payers";
-    private static final String PAYER_API_PATH = PAYERS_API_PATH + "/{payerExternalId}";
+    private static final String PAYER_API_PATH = "/v1/api/accounts/{accountId}/payment-requests/{paymentRequestExternalId}/payers/{payerExternalId}";
 
     private static final Logger LOGGER = PayLoggerFactory.getLogger(PayerResource.class);
-    private final PayerService payerService;
+    private final PaymentProviderFactory paymentProviderFactory;
     private final CreatePayerValidator createPayerValidator = new CreatePayerValidator();
-
-    public PayerResource(PayerService payerService) {
-        this.payerService = payerService;
+    public PayerResource(PaymentProviderFactory paymentProviderFactory) {
+        this.paymentProviderFactory = paymentProviderFactory;
     }
 
     @POST
-    @Path(PAYERS_API_PATH)
+    @Path("/v1/api/accounts/{accountId}/payment-requests/{paymentRequestExternalId}/payers")
     @Consumes(APPLICATION_JSON)
     @Produces(APPLICATION_JSON)
-    public Response createPayer(@PathParam("accountId") Long internalAccountId, @PathParam("paymentRequestExternalId") String paymentRequestExternalId, Map<String, String> createPayerRequest, @Context UriInfo uriInfo) {
+    public Response createPayer(@PathParam("accountId") GatewayAccount gatewayAccount, @PathParam("paymentRequestExternalId") String paymentRequestExternalId, Map<String, String> createPayerRequest, @Context UriInfo uriInfo) {
```

Comment:
_For next PR_,  I assume this is the body as map, we can convert this to actual entities or convert and validate in the resource the map to an instance of PayerRequest (or something like that)

Reply:
yes, planning to do something about that. thanks!